client/src/api/metadata/ApiClient.ts

```diff
@@ -1,18 +1,46 @@
 import axios from 'axios';
 import { MetadataClient } from './MetadataClient';
-import { SigMFMetadata } from '@/Utils/sigmfMetadata';
+import { SigMFMetadata, Annotation, CaptureSegment } from '@/Utils/sigmfMetadata';
 
 export class ApiClient implements MetadataClient {
   async getMeta(account: string, container: string, filePath: string): Promise<SigMFMetadata> {
     const response = await axios.get(`/api/datasources/${account}/${container}/${filePath}`);
-    return response.data;
+    let responseMetaData: SigMFMetadata | null = null;
+    responseMetaData = Object.assign(new SigMFMetadata(), response.data);
+    responseMetaData.annotations = responseMetaData.annotations.map((annotation) =>
+      Object.assign(new Annotation(), annotation)
+    );
+    responseMetaData.captures = responseMetaData.captures.map((capture) =>
+      Object.assign(new CaptureSegment(), capture)
+    );
+    return responseMetaData;
   }
+
   async getDataSourceMeta(account: string, container: string): Promise<SigMFMetadata[]> {
     const response = await axios.get(`/api/datasources/${account}/${container}/meta`);
-    return response.data;
+    let objectFromJson = response.data;
+    let responseMetaData: SigMFMetadata[] = objectFromJson.map((meta) => {
+      let newMeta = Object.assign(new SigMFMetadata(), meta);
+      if (newMeta.annotations) {
+        newMeta.annotations = newMeta.annotations.map((annotation) => Object.assign(new Annotation(), annotation));
```

Comment:
Just a small suggestion but you could avoid the need for the check and the nasted by making the map call 
```suggestion
        newMeta.annotations = newMeta.annotations?.map((annotation) => Object.assign(new Annotation(), annotation));
```